packages/backend/src/engine/backend-secret/backend_secret_fetcher_factory.test.ts

```diff
@@ -25,34 +25,31 @@ void describe('getOrCreate', () => {
   const providerFactory = new BackendSecretFetcherProviderFactory();
   const resourceFactory = new BackendSecretFetcherFactory(providerFactory);
 
+  beforeEach(() => {
+    BackendSecretFetcherFactory.clearRegisteredSecrets();
```

Comment:
Instead of using static methods, we should rather control scope of variables and make the factory an effective singleton in the code, but not in tests.